UI Improvements

jeffmahoney opened the pull request:
This PR contains a few UI improvements:

- In the Shell interface, we don't need to present shell options on operating systems that will not support them. Powershell is now open source and _could_ be on Linux and MacOS even if it's unlikely. Windows CMD will only be on Windows and Bash will not be installed, if it's installed at all, at a UNIX path on Windows. Instead, hide CMD on non-Windows and hide Bash on Windows. Additionally, on non-Windows clients, Bash is the default shell.
- The quarantine functionality is Windows-only ATM due to the artifact used being called `Windows.Remediation.Quarantine`. There's a precondition that ensures it doesn't run on !Windows, but the UI expects the query to run to completion just the same.  Instead, choose an appropriate artifact name based on the client OS and check that it exists. If it does not exist, pop a warning instead of attempting to do it anyway. This also means that adding quarantine support for Linux and MacOS is as simple as adding the appropriate artifact.

jeffmahoney committed:
GUI: Artifact selector (#1790)

* api: add type and description fields to v1/GetArtifacts endpoint

The v1/GetArtifacts endpoint only allows entire artifacts or just the
name to be returned.  In order to make the UI for managing client events
nicer, we'd like to be able to search and present a set of switches to
select the client artifacts.  This change adds the ability to request
the type and description fields.

* api: add type and description fields to v1/GetArtifacts endpoint

The v1/GetArtifacts endpoint only allows entire artifacts or just the
name to be returned.  In order to make the UI for managing client events
nicer, we'd like to be able to search and present a set of switches to
select the client artifacts.  This change adds the ability to request
the type and description fields.

* artifacts: add artifactset parameter type

Historically, the artifacts that handle monitoring of client events for
forwarding to other services have been somewhat clunky to configure.
Originally, it involved an integrated hidden CSV file and a series of
bools that referenced fields within that CSV.  More recently, at least
one has been converted to use a visible CSV that at least gives access
to the CSV editing interface in the GUI.  It's still not a great
interface.

This feature queries the API for a list of artifacts that match the
specified artifact type (or all of them if unspecified) and uses the
list to create a UI consisting of a group of switches (with names and
tooltips describing each one) that can be used to select the artifacts.
It then uses those to assemble the CSV format parameter that the
artifacts already expect so existing configurations will continue to
work without updating.  Since it's CSV behind the scenes, the default
definitions will be consumed as before as well.

To use this interface, an Artifact must be defined using:
type: artifactset

and optionally:
artifact_type: {SERVER,SERVER_EVENT,CLIENT,CLIENT_EVENT,etc}

* Elastic.Events.Client: Update to use new artifactset type

Now that we have the artifactset type in place, we can use it to
implement the Elastic.Events.Client artifact.

* Avoid loading the entire artifact list for each form element.

* sync to master

* fix test

Co-authored-by: Mike Cohen <mike@velocidex.com>

jeffmahoney committed:
shell-viewer: default to Bash on non-Windows clients

While Powershell does exist on Linux/MacOS, Bash makes for a more
reasonable default.  Also, Windows Cmd.exe will never exist there,
so don't present it in the UI.

jeffmahoney committed:
host-info: make quarantine UI more robust with non-Windows client hosts

The quarantine button only activates the Windows.Remediation.Quarantine
artifact which will fail the precondition check on other operating
systems.  The UI will also hang waiting for a completion that never
comes.

This change modifies the behavior such that:
- An appropriate artifact name is chosen based on the client OS
- The presence of that artifact in the repository is checked prior to
  initiating the quarantine operation
- If the artifact does not exist, the quarantine modal displays an alert
  and only allows the user to close the modal

Due to using appropriate artifact names, it follows that quarantine
artifacts for Linux and MacOS can be created and added at any time and
they will immediately become usable.  For other operating systems, more
effort will be required.

guangyee reviewed (approved):
I tested the PR in my local dev environment. It was working as intended.

jeffmahoney committed:
artifactset: add ability to select named sources

A comment on PR #1790 mentioned that there was a shortcoming vs using
CSV fields in that individual sources could not be selected.  This
commit reworks the artifact handling such that it iterates over the
sources and handles named and unnamed sources as needed.

jeffmahoney committed:
Merge pull request #3 from jeffmahoney/sensor-0.6.4/artifact-picker-gui

Sensor 0.6.4/artifact picker gui

jeffmahoney committed:
Merge branch 'SUSE:sensor-base-0.6.4' into sensor-base-0.6.4
